src/components/Heroes/RankColumns.ts

```diff
@@ -0,0 +1,271 @@
+import heroes from 'dotaconstants/build/heroes.json';
+import constants from 'src/components/constants'
+import { abbreviateNumber, displayHeroId } from '../../utility';
+import { GlobalStrings } from 'src/types/common/GlobalStrings'
+import { Heroes } from 'src/types/Hero/Heroes'
+
+type Props = {
+  tabType: HeroesTabType
+  strings: GlobalStrings
+}
+
+enum HeroesTabType {
+  PRO = 'pro',
+  PUBLIC = 'public'
+}
+
+type ProColumn = {
+  displayName: string,
+  field: string,
+  sortFn: boolean,
+  percentBarsWithValue: (row: Row) => string
+}
+
+type PublicColumn = {
+  displayName: string,
+  field: string,
+  sortFn: boolean,
+  percentBarsWithValue: (row: Row) => string
+  displayIcon: string
+  colColor: string
+}
+
+type HeroColumn = {
+  displayName: string,
+  tooltip: string,
+  field: string
+  displayFn: () => JSX.Element,
+  sortFn: (row: Row) => string
+}
+
+type PreparedColumn = (ProColumn | PublicColumn | HeroColumn) & ColumnAdditions
```

Comment:
This looks like a good use case for typescript interfaces, you can declare `Column` interface with all the things that are common & extend it for specific cases

Reply:
Type and interface are exactly the same thing except for syntax. The above is the exact same as had you defined it an interface and extended ProColumn with ColumnAddition